compiler/job.go

```diff
@@ -55,32 +55,18 @@ func NewJob(octx *op.Context, in ast.Seq, src *data.Source, head *lakeparse.Comm
 		octx:      octx,
 		builder:   kernel.NewBuilder(octx, src),
 		optimizer: optimizer.New(octx.Context, src),
-		readers:   readers,
+		reader:    reader,
 		entry:     entry,
 	}, nil
 }
 
-func buildFrom(op ast.Op, head *lakeparse.Commitish) (*ast.From, []*kernel.Reader, error) {
-	var readers []*kernel.Reader
+func buildFrom(op ast.Op, head *lakeparse.Commitish) (*ast.From, *kernel.Reader, error) {
+	var readers *kernel.Reader
```

Comment:
Nit: Move this into the default case since that's the only scope in which it's used.